pkg/logs/input/listener/udp.go

```diff
@@ -72,16 +84,21 @@ func (l *UDPListener) newUDPConnection() (net.Conn, error) {
 
 // read reads data from the tailer connection, returns an error if it failed and reset the tailer.
 func (l *UDPListener) read(tailer *Tailer) ([]byte, error) {
-	inBuf := make([]byte, 4096)
-	n, err := tailer.conn.Read(inBuf)
+	frame := make([]byte, l.frameSize)
+	n, err := tailer.conn.Read(frame)
 	switch {
-	case err != nil && l.isClosedConnError(err):
+	case err != nil && isClosedConnError(err):
 		return nil, err
 	case err != nil:
 		go l.resetTailer()
 		return nil, err
 	default:
-		return inBuf[:n], nil
+		if n == l.frameSize {
+			// Adding '\n' ensures that the current message will be truncated by the decoder,
+			// otherwise it would be mixed the following one.
+			frame[l.frameSize-1] = '\n'
```

Comment:
This will have the edge cases of truncating logs that are exactly the buffer size, but we should be fine with that